File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        cycleSort1toN(nums);
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != i) return i;
        }
        return nums.length; // If all numbers are in place, missing number is n
    }

    static void cycleSort1toN(int[] arr) {
        int n = arr.length;
        int i = 0;
        while (i < n) {
            int element = arr[i];
            if (element >= n || element < 0 || arr[element] == element) {
                i++;
            } else {
                int temp = arr[element];
                arr[element] = element;
                arr[i] = temp;
            }
        }
    }
}

File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int totalSize = nums.size();
        int xor1 = 0;   //array element xor
        int xor2 = 0;   //full elements xor

        for(int i = 0;i<totalSize;i++){
            xor1 = xor1 ^ nums[i];
            xor2 = xor2 ^ i+1;
        }
        return xor1 ^ xor2;
    }
};